shell/common/rasterizer.cc

```diff
@@ -312,29 +294,6 @@ RasterStatus Rasterizer::DoDraw(
     );
   }
 
-  // Pipeline pressure is applied from a couple of places:
-  // rasterizer: When there are more items as of the time of Consume.
-  // animator (via shell): Frame gets produces every vsync.
-  // Enqueing here is to account for the following scenario:
-  // T = 1
-  //  - one item (A) in the pipeline
-  //  - rasterizer starts (and merges the threads)
-  //  - pipeline consume result says no items to process
-  // T = 2
-  //  - animator produces (B) to the pipeline
-  //  - applies pipeline pressure via platform thread.
-  // T = 3
-  //   - rasterizes finished (and un-merges the threads)
-  //   - |Draw| for B yields as its on the wrong thread.
-  // This enqueue ensures that we attempt to consume from the right
-  // thread one more time after un-merge.
-  if (raster_thread_merger_) {
-    if (raster_thread_merger_->DecrementLease() ==
-        fml::RasterThreadStatus::kUnmergedNow) {
-      return RasterStatus::kEnqueuePipeline;
-    }
```

Comment:
Is this really no longer needed? While we've eliminated pipeline pressure, have we really eliminated the need to force a rasterization when the threads merge? I don't see how the layer_tree_holder affects that dynamic. @chinmaygarde any thoughts?

Reply:
I will revert this removal. I think we still need this. I'll add a test that fails if someone tries to do it again.